horovod/common/operations.cc

```diff
@@ -776,9 +766,9 @@ void PerformOperation(TensorTable& tensor_table, MPIResponse response) {
 
     // Tensors may have different first dimension, so we need to use
     // MPI_Allgatherv API that supports gathering arrays of different length.
-    ACTIVITY_START_ALL(entries, timeline, "MPI_ALLGATHER")
-    int* recvcounts = new int[tensor_sizes.size()];
-    int* displcmnts = new int[tensor_sizes.size()];
+    ACTIVITY_START_ALL(entries, timeline, MPI_ALLGATHER)
+    auto* recvcounts = new int[tensor_sizes.size()];
```

Comment:
Is `auto*` different from `auto` here?  I've never seen that.

Reply:
Should be the same, except it highlights ["pointerness"](https://stackoverflow.com/questions/34560365/c11-auto-declaration-with-and-without-pointer-declarator).